File: solutions/jose-rivas/python/p-03-strings/exercise-01.py
try:
    texte = input("Entrez une chaine : ")

    if len(texte) < 1:
        print("Erreur : la chaine doit contenir au moins un caractere.")
    else:
        resultat = ""

        for i in range(len(texte) - 1, -1, -1):
            resultat = resultat + texte[i]

        print("Chaine inversee :", resultat)

except:
    print("Erreur lors de la lecture de la chaine.")

File: solutions/jose-rivas/java/p-03-strings/exercise-01.java
import java.util.Scanner;

public class ReverseString {
    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        try {
            System.out.print("Entrez une chaine : ");
            String texte = scanner.nextLine();

            if (texte.length() < 1) {
                System.out.println("Erreur : la chaine doit contenir au moins un caractere.");
                return;
            }

            String resultat = "";

            for (int i = texte.length() - 1; i >= 0; i--) {
                resultat = resultat + texte.charAt(i);
            }

            System.out.println("Chaine inversee : " + resultat);

        } catch (Exception e) {
            System.out.println("Erreur lors de la lecture de la chaine.");
        }

        scanner.close();
    }
}